File: Application/Client/src/Models/menu.ts
import { Meal } from "./meal";

export interface Menu {
  id?: number;
  date: string;
  isRestauranOpen: boolean;
  meals: Meal[];
}

export interface MenuDto {
  id?: number;
  dayOfWeek: string;
  isRestaurantOpen: boolean;
  meals: MenuMealDto[];
}

export interface MenuMealDto {
  id: number;
  name: string;
  price: number;
  products: string[];
  allergens: string[];
  description: string;
  imageUrls: string[];
}

export interface UpdateRestaurantOpenDto {
  isOpen: boolean;
}

export interface AddMealToMenuDto {
  mealId: number;
}

export interface UpdateMenuMealsDto {
  mealIdsToAdd: number[];
  mealIdsToRemove: number[];
}

File: Application/Web-Based Major Project - API/Web-Based Major Project - API/Entities/Menu.cs
namespace Web_Based_Major_Project___API.Entities
{
    public class Menu
    {
        public int Id { get; set; }
        public DayOfWeek DayOfWeek { get; set; }
        public bool IsRestaurantOpen { get; set; }
        public List<MenuMeal> MenuMeals { get; set; } = new List<MenuMeal>();
    }

    public class MenuMeal
    {
        public int MenuId { get; set; }
        public Menu Menu { get; set; }

        public int MealId { get; set; }
        /*public Meal Meal { get; set; }*/
    }
}
